packages/cli/src/cmds/agentInstaller/projectConfiguration.ts

```diff
@@ -125,7 +125,13 @@ async function getYarnSubprojectsVersionOne(
   subprojects: ProjectConfiguration[]
 ) {
   const cmd = new CommandStruct('yarn', ['workspaces', 'info', '--json'], installer.path);
-  const output = await run(cmd);
+  let output: CommandReturn;
+  try {
+    output = await run(cmd);
+  } catch (e) {
+    if (ChildProcessError.check(e)) return;
+    throw e;
+  }
```

Comment:
Since control flow returns or stops on an exception, I think line 140 can be removed.  This line was trying to parse the string `error` and return.
```
if (line.startsWith('error')) return;
```

Similarly, I think the existing test `yarn getYarnSubprojects for yarn 1, error` can be removed.  It was expecting to parse the string `error`, which isn't produced as output.  And the new test you added seems to cover this.

Reply:
Good point, thanks.